src/main/java/scribe_java/gatekeeper/GatekeeperOAuth2AccessToken.java

```diff
@@ -0,0 +1,49 @@
+package scribe_java.gatekeeper;
+
+import com.github.scribejava.core.model.OAuth2AccessToken;
+import java.util.Objects;
+
+public class GatekeeperOAuth2AccessToken extends OAuth2AccessToken {
+
+    private final String userId;
+
+    public GatekeeperOAuth2AccessToken(String accessToken, String openIdToken, String rawResponse) {
+        this(accessToken, null, null, null, null, openIdToken, rawResponse);
+    }
+
+    public GatekeeperOAuth2AccessToken(String accessToken, String tokenType, Integer expiresIn, String refreshToken,
+                                       String scope, String userId, String rawResponse) {
+        super(accessToken, tokenType, expiresIn, refreshToken, scope, rawResponse);
+        this.userId = userId;
+    }
+
+    public String getUserId() {
+        return userId;
+    }
+
+
+    @Override
+    public int hashCode() {
+        int hash = super.hashCode();
+        hash = 37 * hash + Objects.hashCode(userId);
```

Comment:
This hurts my soul
`return Objects.hashCode(userId, super.hashCode());` instead

Reply:
FIxed